woocommerce/woocommerce
# Reviews block: Convert editor-block.js to editor-block.tsx and replace propTypes with TypeScript definitions

### Changes proposed in this Pull Request:
As part of the effort to convert the codebase to Typescript, this PR converts the `editor-block.js` file from the Reviews block to TypeScript.





Closes #42294 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Login to the dashboard.
2. Create a new post (Posts > Add new post)
3. In the editor, you should see a block inserter icon or an option to "Add Block". Click on it.
4. In the block inserter, search for "Reviews" or browse through the available blocks until you find it.
    ![CleanShot 2024-02-14 at 17 37 01@2x](https://github.com/woocommerce/woocommerce/assets/20469356/f40ddfdf-cc93-4a6f-ba51-8a1a65c6c76c)
6. Click on the "Reviews by Category", "Reviews by Product," and "All Reviews" blocks to add them to your content.
7. Using the Category Selector, select a category for the Reviews By Category block;
8. Using the Product selector, select a product for the Reviews by Product block;
9. Make sure all the blocks are working correctly (test with products with no reviews and products with reviews).
10. Save the post and visit it on the frontend to make sure they are working as expected.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 
Convert editor-block.js from the Reviews block to editor-block.tsx and replace propTypes with TypeScript definitions.

</details>
